# add exceptions

currently the dlogger module does not throw exception when invalid values
are passed to some of its functions.  exceptions should be added where
appropriate.
